### PR TITLE
Test Trigger Modal - debugging

### PR DESCRIPTION
**Pull request by ainouzgali**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature
- **Why was this change needed?** (You can also link to an open issue here)
Add option to test trigger in create/edit template screen.

https://www.loom.com/share/6351d28886664e519da29fc36ad1975b

- **Other information**:
Still need to add tests +docs and fix some styles

### CODE REVIEW

**Review by davidsoderberg — APPROVED** [review]
I think this looks awesome 🏅

File: apps/web/src/pages/templates/editor/TemplateEditorPage.tsx [hunk 1]
```diff
@@ -50,8 +54,32 @@ export default function TemplateEditorPage() {
...
+
+  const onConfirmSaveChanges = async () => {
+    await handleSubmit(onSubmit)();
+    closeSaveChangesModal();
+    openTestWorkflowModal();
+  };
+
+  const onTestWorkflowClicked = async () => {
+    if (isDirty) {
+      openSaveChangesModal();
```
Comment (davidsoderberg) [inline apps/web/src/pages/templates/editor/TemplateEditorPage.tsx:78]: should we return here?

File: apps/web/src/components/templates/SaveChangesModal.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  isVisible,
+  onDismiss,
+  onConfirm,
+}: {
+  isVisible: boolean;
+  onDismiss: () => void;
+  onConfirm: () => void;
+}) {
+  return (
+    <>
```
Comment (davidsoderberg) [inline apps/web/src/components/templates/SaveChangesModal.tsx:14]: This fragment is not needed?

File: apps/web/src/components/templates/SaveChangesModal.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <Modal opened={isVisible} title={<Title size={2}>Save template</Title>} onClose={onDismiss}>
+        <div>
```
Comment (davidsoderberg) [inline apps/web/src/components/templates/SaveChangesModal.tsx:16]: Is this div needed?

File: apps/web/src/pages/templates/editor/TemplateEditorPage.tsx (hunk 1 above)
Comment (ainouzgali) [reply apps/web/src/pages/templates/editor/TemplateEditorPage.tsx:78]: ohh right. oops. It opens one of the modals. I'll add an else. And remove the async

File: apps/web/src/components/templates/TestWorkflowModal.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        label="to:"
+        value={toValue}
+        onChange={setToValue}
+        minRows={3}
+      />
+      <JsonInput
+        formatOnBlur
+        autosize
+        styles={inputStyles}
+        label="payload:"
```
Comment (scopsy) [inline apps/web/src/components/templates/TestWorkflowModal.tsx:89]:
```suggestion
        label="Payload"
```

File: apps/web/src/components/templates/TestWorkflowModal.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      onClose={onDismiss}
+      opened={isVisible}
+      title={<Title>Test Trigger </Title>}
+      data-test-id="test-trigger-modal"
+    >
+      <JsonInput
+        formatOnBlur
+        autosize
+        styles={inputStyles}
+        label="to:"
```
Comment (scopsy) [inline apps/web/src/components/templates/TestWorkflowModal.tsx:80]: To

Comment (scopsy) [inline apps/web/src/components/templates/TestWorkflowModal.tsx:80]:
```suggestion
        label="To"
```

File: apps/web/src/components/templates/TestWorkflowModal.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        label="payload:"
+        value={payloadValue}
+        onChange={setPayloadValue}
+        minRows={3}
+      />
+      <JsonInput
+        formatOnBlur
+        autosize
+        styles={inputStyles}
+        label="overrides:"
```
Comment (scopsy) [inline apps/web/src/components/templates/TestWorkflowModal.tsx:98]:
```suggestion
        label="Overrides (optional)"
```

File: apps/web/src/components/templates/TestWorkflowModal.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+  const toTrigger = `{ 
+    ${subscriberVariables
+      ?.map((variable) => {
+        return `"${variable.name}": "${
+          (currentUser && currentUser[variable.name === 'subscriberId' ? 'id' : variable.name]) ?? 'REPLACE_WITH_DATA'
+        }"`;
+      })
+      .join(',\n    ')}
+}`;
```
Comment (scopsy) [inline apps/web/src/components/templates/TestWorkflowModal.tsx:34]: Can we add component tests for this modal to make sure all those variables are generated properly? Also looks like this behaviour is duplicated from the curl section of the trigger. Maybe we can reuse the code for both?

**Review by p-fernandez — APPROVED** [review]
🌟

File: apps/web/src/components/templates/TestWorkflowModal.tsx (hunk 7 above)
Comment (ainouzgali) [reply apps/web/src/components/templates/TestWorkflowModal.tsx:34]: I feel that reusing will look more awkward than not. As in the trigger snippet it is one block of code with extra code lines (like curl address or the novu import for example) and here we also substitute part of the variables to their actual values. But I get what you mean.

**Pull request merged**